hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.recordstore;
+
+import com.hazelcast.map.impl.SizeEstimator;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Represents actual storage layer behind a {@link RecordStore}.
+ * Includes basic storage operations.
+ *
+ * @param <K> the key type to be put in this storage.
+ * @param <V> the value type to be put in this storage.
+ */
+public interface Storage<K, V> {
+
+    void put(K key, V record);
+
+    void updateRecordValue(K key, V record, Object value);
+
+    V get(K key);
+
+    Object remove(K key);
+
+    boolean containsKey(K key);
+
+    Collection<V> values();
+
+    Set<K> keySet();
+
+    int size();
+
+    boolean isEmpty();
+
+    void clear();
+
+    void destroy();
+
+    SizeEstimator getSizeEstimator();
+
+    void setSizeEstimator(SizeEstimator sizeEstimator);
```

Comment:
Why do we want to set it? I checked the code and it is only set in the EvictionMaxSizePolicyTest. Since it is just used for testing, don't expose it through your API. BUt just do a hard cast and create a setter on the implementation if needed.